hivemq-edge/src/main/java/com/hivemq/bridge/mqtt/RemoteMqttForwarder.java

```diff
@@ -194,32 +195,53 @@ private PUBLISH convertPublishAfterBridge(@NotNull PUBLISH publish, int hopCount
         mqtt5Builder.withOnwardQos(modifiedQoS);
         mqtt5Builder.withRetain(localSubscription.isPreserveRetain() && publish.isRetain());
         mqtt5Builder.withUserProperties(convertUserProperties(publish.getUserProperties(), hopCount));
-        PUBLISH newPublish = mqtt5Builder.build();
-        return newPublish;
+        return mqtt5Builder.build();
     }
 
+
     @SuppressWarnings("ResultOfMethodCallIgnored")
-    private void sendPublishToRemote(
+    private synchronized void sendPublishToRemote(
             @NotNull PUBLISH publish, @NotNull String queueId, @NotNull PUBLISH origPublish) {
         final Mqtt5Publish mqtt5Publish = convertPublishForClient(publish);
```

Comment:
Improvement: The convert could be moved after the following if block, in the case that the bridge is not connected it might be ran twice otherwise